### PR TITLE
C175330 cleanup migration coord 4.0

### PR DESCRIPTION
**Pull request by jonmeredith**
See [CASSANDRA-17533](https://issues.apache.org/jira/browse/CASSANDRA-17533)

### CODE REVIEW

File: test/distributed/org/apache/cassandra/distributed/action/GossipHelper.java [hunk 1]
```diff
@@ -222,7 +223,7 @@ public void accept(IInvokableInstance pullTo)
                 InetAddressAndPort endpoint = toCassandraInetAddressAndPort(pullFrom);
                 EndpointState state = Gossiper.instance.getEndpointStateForEndpoint(endpoint);
                 MigrationCoordinator.instance.reportEndpointVersion(endpoint, state);
-                MigrationCoordinator.instance.awaitSchemaRequests(TimeUnit.SECONDS.toMillis(10));
+                assertTrue(MigrationCoordinator.instance.awaitSchemaRequests(TimeUnit.SECONDS.toMillis(10)));
```
Comment (dcapwell) [inline test/distributed/org/apache/cassandra/distributed/action/GossipHelper.java]: plz be kind and add a message ^_^

**Review by dcapwell — APPROVED** [review]
approved, one non-blocking plz do

File: test/distributed/org/apache/cassandra/distributed/impl/Instance.java [hunk 2]
```diff
@@ -541,7 +541,7 @@ public void startup(ICluster cluster)
                 StorageService.instance.registerDaemon(CassandraDaemon.getInstanceForTesting());
                 if (config.has(GOSSIP))
                 {
-                    MigrationManager.setUptimeFn(() -> TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startedAt));
+                    MigrationCoordinator.setUptimeFn(() -> TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startedAt));
```
Comment (dcapwell) [inline test/distributed/org/apache/cassandra/distributed/impl/Instance.java]: spoke in slack, can you also fix the startedAt bug?  This is object init, which != uptime.

When we run for 1m and restart, the restart should have uptime=0 but will be uptime=1m due to the fact we don't cleanup the object.

**Review by dcapwell — CHANGES_REQUESTED** [review]

**Review by dcapwell — APPROVED** [review]

**Comment by jonmeredith** [comment]
Reviewed, starting merge. Thanks for the review.

**Pull request closed without merging**